### PR TITLE
Hairstyling update

### PR DESCRIPTION
**Pull request by Mickyan**
## About The Pull Request

Added a few new toys to mess with hair: you can modify wigs by hand and change people's hairstyles with a razor on help intent. Also a new wig type that always matches your hair color.

## Why It's Good For The Game

Random wigs are laaame but they're neat for disguises or just messing around. Think about it: skeletons with afros
Giving people silly hairstyles is more fun than just shaving their head (race/gender restrictions still apply)

## Changelog
:cl:
add: You can now use a razor on help intent to change your fellow crewmembers' hair/facial hair styles
add: Using a wig in your hand lets you change it to a new style/color
add: Natural Wigs are now available in the Clothesmate: made of 100% natural chromatic nanofibers, these wigs automatically match the color of your natural hair
/:cl:

I've been testing this fairly extensively, everything looks in order

The diff for the razor looks really messy but in reality I just added two blocks of code

### CODE REVIEW

**Comment by necromanceranne** [comment]
Finally Greased Lightning, the fastest ethereal chaplain to ever live, will get the pompadour he has sorely deserved but was cruelly denied of him.

**Comment by ATH1909** [comment]
Will you be able to shave someone bald without needing to go though a menu if you use harm/grab/disarm intent?

**Comment by Mickyan** [comment]
yes

**Comment by FantasticFwoosh** [comment]
Allow us to preset, craft or order wigs from cargo please. It is a major inconvenience to not have a wig on hand to cover our bald-spot without pestering the clown. Good changes all around however.

File: code/game/objects/items/cosmetics.dm [hunk 1]
```diff
@@ -131,60 +131,100 @@
...
+					if(do_after(user, 50, target = H))
+						user.visible_message("[user] shaves [user.p_their()] facial hair clean with [src].", \
+											 "<span class='notice'>You finish shaving with [src]. Fast and clean!</span>")
 						shave(H, location)
+				else
+					var/turf/H_loc = H.loc
+					user.visible_message("<span class='warning'>[user] tries to shave [H]'s facial hair with [src].</span>", \
+										 "<span class='notice'>You start shaving [H]'s facial hair...</span>")
+					if(do_after(user, 50, target = H))
+						if(H_loc == H.loc)
```
Comment (ShizCalev) [inline code/game/objects/items/cosmetics.dm]: isn't this already handled by the do_after?

Comment (ShizCalev) [inline code/game/objects/items/cosmetics.dm]: isn't this already handled by do_after?

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/cosmetics.dm]: https://github.com/tgstation/tgstation/blob/513944c8daff765330ffb7173c2882e119dbd58d/code/__HELPERS/mobs.dm#L277

Comment (Mickyan) [reply code/game/objects/items/cosmetics.dm]: I didnt actually write that piece of code but yeah I think so

If you want me to fix it it'll have to wait for after the weekend as i'm away from home

**Comment by PKPenguin321** [comment]
Barber job when?

File: code/game/objects/items/cosmetics.dm [hunk 2]
```diff
@@ -131,60 +131,98 @@
...
+					if(!get_location_accessible(H, location))
+						to_chat(user, "<span class='warning'>The mask is in the way!</span>")
+						return
+					user.visible_message("<span class='notice'>[user] tries to change [H]'s facial hair style using [src].</span>", "<span class='notice'>You try to change [H]'s facial hair style using [src].</span>")
+					if(new_style && do_after(user, 60, target = H))
+						user.visible_message("<span class='notice'>[user] successfully changes [H]'s facial hair style using [src].</span>", "<span class='notice'>You successfully change [H]'s facial hair style using [src].</span>")
+						H.facial_hair_style = new_style
+						H.update_hair()
+
+					return
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/cosmetics.dm]: what's this return for?

File: code/game/objects/items/cosmetics.dm [hunk 3]
```diff
@@ -131,60 +131,98 @@
...
+				var/new_style = input(user, "Select a hair style", "Grooming")  as null|anything in GLOB.hair_styles_list
+				if(!get_location_accessible(H, location))
+					to_chat(user, "<span class='warning'>The headgear is in the way!</span>")
+					return
+				user.visible_message("<span class='notice'>[user] tries to change [H]'s hairstyle using [src].</span>", "<span class='notice'>You try to change [H]'s hairstyle using [src].</span>")
+				if(new_style && do_after(user, 60, target = H))
+					user.visible_message("<span class='notice'>[user] successfully changes [H]'s hairstyle using [src].</span>", "<span class='notice'>You successfully change [H]'s hairstyle using [src].</span>")
+					H.hair_style = new_style
+					H.update_hair()
 				return
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/cosmetics.dm]: same here

File: code/modules/clothing/head/misc_special.dm [hunk 4]
```diff
@@ -189,6 +189,7 @@
 	flags_inv = HIDEHAIR
 	var/hair_style = "Very Long Hair"
 	var/hair_color = "#000"
+	var/fixedcolor = TRUE //can color be changed manually?
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/head/misc_special.dm]: the var is called "fixedcolor" but setting it to TRUE means you can change the color, aka it's not fixed.

File: code/modules/clothing/head/misc_special.dm [hunk 5]
```diff
@@ -216,11 +217,39 @@
 		M.color = hair_color
 		. += M
 
+/obj/item/clothing/head/wig/attack_self(mob/user)
+	var/new_style = input(user, "Select a hair style", "Wig Styling")  as null|anything in (GLOB.hair_styles_list - "Bald")
+	if(!user.canUseTopic(src, BE_CLOSE))
+		return
+	if(new_style)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/head/misc_special.dm]:
```suggestion
	if(new_style && new_style != hair_style)
```

File: code/modules/clothing/head/misc_special.dm [hunk 6]
```diff
@@ -216,11 +217,39 @@
 		M.color = hair_color
 		. += M
 
+/obj/item/clothing/head/wig/attack_self(mob/user)
+	var/new_style = input(user, "Select a hair style", "Wig Styling")  as null|anything in (GLOB.hair_styles_list - "Bald")
+	if(!user.canUseTopic(src, BE_CLOSE))
+		return
+	if(new_style)
+		hair_style = new_style
+		user.visible_message("<span class='notice'>[user] changes the [src]'s hairstyle to [new_style].</span>", "<span class='notice'>You change the [src]'s hairstyle to [new_style].</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/clothing/head/misc_special.dm]:
```suggestion
		user.visible_message("<span class='notice'>[user] changes \the [src]'s hairstyle to [new_style].</span>", "<span class='notice'>You change \the [src]'s hairstyle to [new_style].</span>")
```

File: code/game/objects/items/cosmetics.dm (hunk 2 above)
Comment (Mickyan) [reply code/game/objects/items/cosmetics.dm]: Return after H.update_hair(), I guess I should add an extra tab here

File: code/modules/clothing/head/misc_special.dm (hunk 4 above)
Comment (Mickyan) [reply code/modules/clothing/head/misc_special.dm]: I my mind it was "fixed color" as in "it doesn't change on its own" but I get the confusion, I'll change to adjustablecolor

File: code/game/objects/items/cosmetics.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/cosmetics.dm]: that makes more sense

**Pull request merged**